internal/logic_handlers/categorylogic/adapters.go

```diff
@@ -15,4 +15,6 @@ type persistor interface {
 	GetCategoryTypeById(ctx context.Context, tx persistence.TransactionHandler, id int) (*model.CategoryType, error)
 	GetCategoryByName(ctx context.Context, tx persistence.TransactionHandler, name string) (*model.Category, error)
 	UpdateCategory(ctx context.Context, tx persistence.TransactionHandler, params *model.UpdateCategory) (*model.Category, error)
+	DeleteCategory(ctx context.Context, tx persistence.TransactionHandler, id int) error
+	RestoreCategory(ctx context.Context, tx persistence.TransactionHandler, id int) error
```

Comment:
Ahh.. Remove this as well - make **separate** prs exclusively for:
1. Delete, and Restore category
2. Organisations CRUD